src/Mono.Android/Xamarin.Android.Net/ServerCertificateCustomValidator.cs

```diff
@@ -0,0 +1,185 @@
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Net.Security;
+using System.Security.Cryptography.X509Certificates;
+
+using Javax.Net.Ssl;
+
+using JavaCertificateException = Java.Security.Cert.CertificateException;
+using JavaX509Certificate = Java.Security.Cert.X509Certificate;
+
+namespace Xamarin.Android.Net
+{
+	internal sealed class ServerCertificateCustomValidator
+	{
+		public IHostnameVerifier HostnameVerifier { get; } = new AlwaysAcceptingHostnameVerifier();
```

Comment:
Similarly, this should become an `AlwaysAcceptingHostnameVerifier.Instance` static property; we only ever need *one*, right?

Reply:
A downside with making a static `AlwaysAcceptingHostnameVerifier.Instance` property is that it would never be disposed, meaning if you had an app that set `AndroidMessageHandler.ServerCertificateCustomValidationCallback` to *something*, then later set it to *null*, the `AlwaysAcceptingHostnameVerifier` would still be kept alive.

Is this a scenario to worry about, though?  If you're setting `AndroidMessageHandler.ServerCertificateCustomValidationCallback`, why would you clear it?  I don't understand the use case.

Reply:
I don't think we need to worry about the scenario where you set the callback to null. On the other hand, it's possible that the whole message handler is collected and the hostname verifier will be kept alive. We could add some instance counting and remove the hostname when the last validator instance is finalized but that seems like an overkill.

The official recomendation is to instantiate HttpClient (and transitively the HTTP message handlers) once and then use it for the lifetime of the application (https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclient?view=net-6.0#instancing). The documentation also suggests creating multiple HttpClients when the app needs to make requests with different setting and I don't think we need to optimize for the case when the `ServerCertificateCustomValidationCallback` property is set multiple times (or set to `null`).

Reply:
> Similarly, this should become an `AlwaysAcceptingHostnameVerifier.Instance` static property; we only ever need _one_, right?

Yeah, but it would also add initialization pressure to apps which don't use the validator? Even if they don't access the property, the static constructor would run anyway?

Reply:
If the app doesn't use the validator, then all of this code is linked away, so I don't think it's a problem.